heat_pump_adoption_modelling/analysis/MCS/plot_mcs.py

```diff
@@ -0,0 +1,549 @@
+# File: heat_pump_adoption_modelling/analysis/MCS/plot_mcs.py
+"""Produce plots relating to heat pump costs/capacities:
+- Counts by manufacturer
+- Median costs by year and tech type
+- Mean ASHP SCOP by year and flow temp
+- Costs by ASHP capacity
+- ASHP capacities by property archetype
+- ASHP costs by property archetype
+
+Current draft of paper can be found here: https://docs.google.com/document/d/1Es-eWt5kLTbsZRtazi23IbpaPIdPDe29juqWGnAGKHc/edit?usp=sharing
+"""
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+from heat_pump_adoption_modelling import PROJECT_DIR
+from heat_pump_adoption_modelling.getters.load_mcs import load_inflation
+from heat_pump_adoption_modelling.pipeline.preprocessing.mcs_epc_joining import (
+    join_mcs_epc_data,
+)
+
+# Load data for plotting
+
+
+def plottable_data(file_path=PROJECT_DIR / "outputs/mcs_epc.csv"):
+    """Load merged MCS/EPC data in a suitable form for plotting
+    (with additional variables: "built_post_1950" indicating
+    whether or not the property was built after 1950, and
+    "inflated_cost" indicating inflation-adjusted installation cost).
+
+    Parameters
+    ----------
+    file_path : str
+        Path to saved merged data.
+        If none, data is loaded from scratch.
+
+    Return
+    ----------
+    merged : pd.DataFrame
+        Merged MCS-EPC data with added variables.
+    """
+
+    if file_path is None:
+        merged = join_mcs_epc_data()
+    else:
+        merged = pd.read_csv(file_path)
+
+    # Add inflated cost variable
+    inflation = load_inflation()
+    merged = merged.merge(inflation, how="left", on="year")
+    merged["inflated_cost"] = merged["cost"] * merged["multiplier"]
+
+    # Add indicator variable for properties built after 1950
+    ages_dict = {
+        "England and Wales: before 1900": False,
+        "1900-1929": False,
+        "1930-1949": False,
+        "1950-1966": True,
+        "1965-1975": True,
+        "1976-1983": True,
+        "1983-1991": True,
+        "1991-1998": True,
+        "1996-2002": True,
+        "2003-2007": True,
+        "2007 onwards": True,
+        "unknown": pd.NA,
+        np.nan: pd.NA,
+        None: pd.NA,
+    }
+
+    merged["built_post_1950"] = merged["CONSTRUCTION_AGE_BAND"].apply(
+        lambda x: ages_dict[x]
+    )
+
+    return merged
+
+
+# Define functions to assist with "archetype" boxplots
+
+
+def bool_not(series):
+    """Given a series of bools and NAs, return a series
+    with NAs in the same places but bools inverted.
+
+    Parameters
+    ----------
+    series : pd.Series
+        A series of bools and NAs.
+
+    Return
+    ----------
+    result : pd.Series
+        A series with NAs in the same places and bools inverted.
+    """
+
+    # Initialise with a series of NAs
+    result = pd.Series(np.repeat(pd.NA, len(series)))
+
+    result.loc[series == True] = False
+    result.loc[series == False] = True
+
+    return result
```

Comment:
I understand the problem but you may be able to solve this much quicker and more elegantly. Please check again if this really does what it should but I'd recommend something along those lines (should work with Series as well). 

![Screenshot 2021-12-03 at 15 04 04](https://user-images.githubusercontent.com/42718928/144624851-5d0ce5b0-7d45-4898-be61-74abcb2857cf.png)


Alternatively, I would recommend a dict mapping True to False, False to True and NaN to NaN. That should be much faster.

If the <NA> instead of NaN worries you (not quite sure if it should), restore the Nan with the following command: 

![Screenshot 2021-12-03 at 15 10 25](https://user-images.githubusercontent.com/42718928/144625620-a51184fa-5e27-4cba-b74b-f523f84f3e40.png)

Reply:
Perfect, this is great! I've replaced the bool_not function with this